### PR TITLE
test: cypress setup and add login ui test

### PR DESCRIPTION
**Pull request by patelami**
This branch includes cypress ui tests POC:

- Cypress setup 
- Login ui test

### CODE REVIEW

File: gravitee-am-cypress/cypress/integration/login-ui-test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    cy.setCookies();
+  });
+
+  it("should load login page", () => {
+    cy.url().should("contain", "login");
+  });
+
+  it(`should have login page elements`, () => {
+    cy.get(loginFormTitle).should("be.visible");
+    cy.get(loginFormTitle).contains("Sign In");
```
Comment (ctschacher) [inline gravitee-am-cypress/cypress/integration/login-ui-test.ts:40]: It's not wrong but just a hint if you test the same element again you can also use 'and' to combine them:
`cy.get(loginFormTitle).should("be.visible").and('contain', 'Sign In');`
 The result is the same as Cypress will perform two separate tests it's maybe just more readable.

File: gravitee-am-cypress/cypress/integration/login-ui-test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    cy.get(loginFormSubTitle).should("be.visible");
+    cy.get(loginFormLogo).should("be.visible");
+  });
+
+  it(`should sucessful login`, () => {
+    cy.get(userName).type("admin");
+    cy.get(password).type("adminadmin");
```
Comment (ctschacher) [inline gravitee-am-cypress/cypress/integration/login-ui-test.ts:47]: In APIM we have users defined as environment variables in cypress.json, could be an option here as well.

File: gravitee-am-cypress/cypress/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "lib": ["es5", "dom"],
```
Comment (ctschacher) [inline gravitee-am-cypress/cypress/tsconfig.json:4]: ES5 is quite outdated, ES2020 is maybe more appropriate.

File: gravitee-am-cypress/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "private": true,
+    "description": "Gravitee.io AM - Cypress",
+    "scripts": {
+        "test": "cypress open",
+        "test:bulk": "cypress run --config-file cypress/integration/bulk/cypress-config.json",
+        "test:rest-api": "cypress run --config-file cypress/integration/rest-api/cypress-config.json",
+        "test:e2e": "cypress open --config-file cypress/integration/e2e/cypress-config.json",
+        "test:ui": "cypress open --config-file cypress/integration/ui/cypress-config.json",
+        "test:dev": "cypress run --env printLogsToConsole=always --config-file cypress/integration/rest-api/cypress-config.json",
+        "test:parallel": "cypress run --record --group apis --spec 'cypress/integration/rest-api/apis/**/*' --config-file cypress/integration/rest-api/cypress-config.json",
```
Comment (ctschacher) [inline gravitee-am-cypress/package.json:13]: The files and folders for these commands don't exist (yet?), so they won't work.

File: .gitignore [hunk 5]
```diff
@@ -27,3 +27,8 @@ gravitee-am-management-api/gravitee-am-management-api-standalone/gravitee-am-man
 dependency-reduced-pom.xml
 audit-logs
 gravitee-am-ui/.vscode/
+
+# Cypress
+gravitee-apim-cypress/cypress/videos/
+gravitee-apim-cypress/cypress/screenshots/
```
Comment (leleueri) [inline .gitignore:33]: I think you mean gravitee-am-cypress :)

File: gravitee-am-cypress/cypress.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "baseUrl": "http://localhost/am/ui",
```
Comment (leleueri) [inline gravitee-am-cypress/cypress.json:2]: Is it possible to move the baseUrl into env section?
Will be probably more flexible and allow us to run easily the tests on multiple platform.

**Review by leleueri — CHANGES_REQUESTED** [review]

**Review by leleueri — CHANGES_REQUESTED** [review]
LGTM but circleci fails due to license header issue.

**Pull request merged**